### PR TITLE
Fix crashes in gltf import with more than one morph target

### LINKED ISSUE #16124: Godot crashes during import of GLTF2.0 with two and more morph shapes
**Godot version:**
Godot 3.0 RC3

**OS/device including version:**
GNU/Linux lubuntu 16.04 x86_64 with Linux 4.6.4-040604-generic

**Issue description:**
Export file with one morph shape from blender to gltf 2.0, and it will import with no problem.
But if you add another shape, godot will crash during the import.

**Steps to reproduce:**
https://youtu.be/VkV-OfE2dus

**Minimal reproduction project:**
[godot3.0_morph_bug.blend.zip](https://github.com/godotengine/godot/files/1670655/godot3.0_morph_bug.blend.zip)

### PR DESCRIPTION
**Pull request by pouleyKetchoupp**
Work around an issue in Blender gltf 2.0 exporter.

More details in my comment:
https://github.com/godotengine/godot/issues/16124#issuecomment-556222555

Fixes #16124

### CODE REVIEW

File: editor/import/editor_scene_importer_gltf.cpp [hunk 1]
```diff
@@ -2395,7 +2398,43 @@ Error EditorSceneImporterGLTF::_parse_animations(GLTFState &state) {
 
 				track->weight_tracks.resize(wc);
 
-				const int wlen = weights.size() / wc;
+				int wlen = weights.size() / wc;
+
+				const int expectedLen = times.size() * outputElements;
```
Comment (akien-mga) [inline editor/import/editor_scene_importer_gltf.cpp]: We use `snake_case` for identifiers, so these should be `expected_len` and `output_elements`.

**Comment by akien-mga** [comment]
CC @fire @marstaik

File: editor/import/editor_scene_importer_gltf.cpp (hunk 1 above)
Comment (pouleyKetchoupp) [reply editor/import/editor_scene_importer_gltf.cpp]: Oops, sorry for that.

**Comment by fire** [comment]
I don't know if this workaround should be in our code.

1. The code changes hard codes multiple constants
2. Not sure about compatibility with maya2gltf's gltf2 export.

We should not crash and error at the first invalid access. Is that possible?

**Comment by pouleyKetchoupp** [comment]
Thanks for your feedback.

We could just return a parse error when this case happens (like this code does in the else case). Then it means scenes with multiple morph targets are not going to work for Blender users if they use any version under 2.81 (which is not released yet).

For the two other comments:
1. These constants just mean that a spline interpolation requires 3 values while a linear interpolation requires only one value. I can add comments to make it clearer but I really don't think it's that bad.
2. I can't use Maya's gltf exporter, though I would be happy to test an example scene if someone can make one with at least two morph targets. If this exporter follows the specs, it should still work with no problem.

**Comment by fire** [comment]
I've been advertising Blender 2.81 and Godot 3.2 master as the "working" versions, so I'm not that interested in back porting support for Blender 2.80.

Also Blender 2.81 https://twitter.com/blender_org/status/1197193642189959169?s=20 was announced to release today.

Thoughts?

**Comment by pouleyKetchoupp** [comment]
Ok, that makes a lot of sense!

I'll make a separate PR just for the error instead of crash. Let's close this one.

**Pull request closed without merging**